more_itertools/more.py

```diff
@@ -3956,6 +3957,60 @@ def combination_index(element, iterable):
     return factorial(n + 1) // (factorial(k + 1) * factorial(n - k)) - index
 
 
+def combination_with_replacement_index(element, iterable):
+    """Equivalent to ``list(combinations_with_replacement(iterable, r)).index(element)``
+
+    The subsequences with repetition of *iterable* that are of length *r* can
+    be ordered lexicographically. :func:`combination_with_replacement_index`
+    computes the index of the first *element*, without computing the previous
+    combinations with replacement.
+
+        >>> combination_with_replacement_index('adf', 'abcdefg')
+        20
+
+    ``ValueError`` will be raised if the given *element* isn't one of the
+    combinations with replacement of *iterable*.
+    """
+    l = ilen(element)
```

Comment:
Could you explain why you're using `ilen` here? If the element is an iterator, then this will exhaust it. Would `len` make more sense?

Reply:
This should have been a normal `len`. I have changed it by converting `element` to a `tuple` and then fetching the length using `len`.